src/System.Threading.Timer/tests/TimerFiringTests.cs

```diff
@@ -106,15 +107,13 @@ public void MultipleTimers_ShouldFire_InCorrectOrder()
         using (t2 = new Timer(tc, t2State, TimeSpan.FromMilliseconds(0), TimeSpan.FromMilliseconds(-1)))
         {
             // Wait for both events to fire
-            Assert.True(are.WaitOne(TimeSpan.FromMilliseconds(1500 /*1.5 seconds*/)));
+            Assert.True(are.WaitOne(TimeSpan.FromMilliseconds(3000 /*3 seconds*/)));
```

Comment:
For the cases of ```Assert.True(are.WaitOne(...))``` where we expect the event to be set and the only reason we have a timeout at all is to avoid hanging the tests for a prolonged period of time (Jenkins will wait a few hours to kill a hung test), I'd suggest having a constant ExpectedCallbackWaitTimeout, or something like that, which is used across all of the tests, and which is set to 30 seconds or something obscenely long.